### PR TITLE
Add form submission to Slider.Modify.

### PR DESCRIPTION
**Pull request by davywentworth**

### CODE REVIEW

File: js/Source/Slider.Modify/Behavior.Slider.Modify.js [hunk 1]
```diff
@@ -61,6 +64,13 @@ Behavior.addGlobalFilter('Slider.Modify', {
         roundAfterSnap: api.getAs(Number, 'roundAfterSnap')
       }
     );
+    if(formTarget){
+      var timer;
+      slider.addEvent('move', function(){
+        clearTimeout(timer);
+        timer = setTimeout(formTarget.submit, api.get('formSubmitDelay'));
+      });
+    }
```
Comment (anutron) [inline js/Source/Slider.Modify/Behavior.Slider.Modify.js]: I think all of this should be a Behavior Plugin. I.e. if you have an app that has a slider that doesn't need this functionality, you don't need to include it. Plugins are pretty easy; you're handed the instance the filter created (in this case, the slider). Your options would be on the plugin (`formToSubmit`, `formSubmitDelay`). It keeps this behavior clean and single purpose and your plugin is nice and tidy. Give it a shot!

File: sandbox/docs/JavaScript/Forms/Behavior.Slider.Modify.html.erb [hunk 2]
```diff
@@ -228,6 +228,24 @@ Slider multiplied by 7 (setting the html property is the default behavior): <spa
...
+        </tr>
+        <tr>
+          <td>formSubmitDelay</td>
+          <td>Number</td>
+          <td><code>1000</code></td>
+          <td>
+            If <code>formToSubmit</code> is present, this is the amount of delay
+            before that form is submitted after each <code>move</code> event.
+          </td>
+        </tr>
```
Comment (anutron) [inline sandbox/docs/JavaScript/Forms/Behavior.Slider.Modify.html.erb]: I would add a separate table in the docs for the plugin; no need for a new page of its own.

File: js/Source/Slider.Modify/Behavior.Slider.Modify.js (hunk 1 above)
Comment (anutron) [inline js/Source/Slider.Modify/Behavior.Slider.Modify.js]: Examples you can find in our codebase:
* Behavior.InteractiveMessageList
* Behavior.BS.FormValidator
* Behavior.BS.Popup.FormRequest

Note that the `api` object the plugin is handed is *for the plugin* so if you have a behavior called `Foo` and a plugin called `Foo.Bar` the api instance handed to the plugin knows how to read `data-foo-bar-options`. If you want to read options from the `foo` portion of the element, you have to create your own API instance, which is easy:

`var formRequestAPI = new BehaviorAPI(element, 'foo');`

You can see an example of this in `Behavior.BS.Popup.FormRequest`

**Comment by davywentworth** [comment]
I don't understand why a plugin is necessary. why wouldn't this form-input-like thing be able to submit a form?

**Comment by anutron** [comment]
hrmph. Yeah, you have a point. It just looks so much like a cleanly excisable bit of functionality. But up to you.

**Pull request merged**